pennylane_lightning/core/src/simulators/lightning_qubit/StateVectorLQubitManaged.hpp

```diff
@@ -139,6 +140,39 @@ class StateVectorLQubitManaged final
 
     ~StateVectorLQubitManaged() = default;
 
+    /**
+     * @brief Prepares a single computational basis state.
+     *
+     * @param index Index of the target element.
+     */
+    void setBasisState(const std::size_t index) {
+        std::fill(data_.begin(), data_.end(), 0);
```

Comment:
All good. As long as we build with -O2 and better, and clang/gcc/msvc this should offload to `memset` and be as fast as possible.